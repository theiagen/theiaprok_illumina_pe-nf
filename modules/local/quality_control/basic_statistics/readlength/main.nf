process READLENGTH {
    tag "$meta.id"
    label 'process_low'

    container = 'us-docker.pkg.dev/general-theiagen/staphb/bbtools:38.76'

    input:
    tuple val(meta), path(read1), path(read2)

    output:
    tuple val(meta), path("AVERAGE_READ_LENGTH.txt"), emit: average_read_length
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''

    """

    readlength.sh in=${read1} > STDOUT_FORWARD
    readlength.sh in=${read2} > STDOUT_REVERSE

    avg_forward=\$(cat STDOUT_FORWARD | grep "#Avg:" | cut -f 2)
    avg_reverse=\$(cat STDOUT_REVERSE | grep "#Avg:" | cut -f 2)

    echo "Average read length for forward reads: \$avg_forward"
    echo "Average read length for reverse reads: \$avg_reverse"

    results=\$(awk "BEGIN { printf \\"%.2f\\", (\$avg_forward + \$avg_reverse) / 2 }")

    echo "\$results" > AVERAGE_READ_LENGTH.txt


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bbtools: \$(bbversion.sh)
    END_VERSIONS
    """

    stub:
    """
    echo "1000" > AVERAGE_READ_LENGTH.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bbtools: \$(bbversion.sh)
    END_VERSIONS
    """
}