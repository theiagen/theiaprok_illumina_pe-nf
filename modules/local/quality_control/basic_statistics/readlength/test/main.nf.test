nextflow_process {
    name "Test Process READLENGTH"
    script "../main.nf"
    process "READLENGTH"
    
    test("bbtools readlength test") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test_sample' ],
                    file(params.pipelines_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists: true),
                    file(params.pipelines_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                {  assert snapshot(
                    process.out.average_read_length,
                    process.out.versions
                    ).match() }
            )
        }
    }
    
    test("readlength - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test_sample' ],
                    file(params.pipelines_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists: true),
                    file(params.pipelines_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}