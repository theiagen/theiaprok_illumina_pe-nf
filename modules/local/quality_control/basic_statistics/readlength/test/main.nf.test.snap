{
    "bbtools readlength test": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "AVERAGE_READ_LENGTH.txt:md5,e38877d821e4f2fc156ecae3cd46a212"
                ]
            ],
            [
                "versions.yml:md5,7d351a35582ee9ebc976a098c87f7d0d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-17T17:14:56.480723"
    },
    "readlength - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "AVERAGE_READ_LENGTH.txt:md5,ad865d2f63b9feb2552c220385fbb7e3"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7d351a35582ee9ebc976a098c87f7d0d"
                ],
                "average_read_length": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "AVERAGE_READ_LENGTH.txt:md5,ad865d2f63b9feb2552c220385fbb7e3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7d351a35582ee9ebc976a098c87f7d0d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-17T17:15:08.680067"
    }
}